src/Compilers/Core/Portable/SourceGeneration/GeneratedSourceText.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// A source text created by an <see cref="ISourceGenerator"/>
```

Comment:
>ISourceGenerator [](start = 47, length = 16)

The doc comment for ISourceGenerator doesn't mention this type at all. I think it would be good to have a comment explaining what ISourceGenerator is supposed to do and how it is supposed to do that.